### PR TITLE
Issue #2284 Fix preflight messages around proxy env

### LINKED ISSUE #2284: Improve preflight messages about proxy
### General information

  * Minishift version: Latest master, 166a8eba2d80be143af52d99ffe206488d4033fe

Currently 

### Steps to reproduce

  1. `minishift start --v 5` and check the output
  

### Expected

As I understand the idea is to notify user is the setup is using proxy setting from environment to keep things transparent.  However with current code the message always shows up on the console and there is a bug when `--v 5` is used. 
i.e. `proxy: ""` has an empty value in `Checking if https://github.com is reachable (using proxy: "") ... OK` 

### Actual

```
$ minishift start --v 5
-- minishift version: v1.15.1+80a98444
-- Starting profile 'minishift'
-- Checking if https://github.com is reachable (using proxy: "") ... OK
-- Checking if requested OpenShift version 'v3.9.0' is valid ... OK
```

Without `--v 5`

```
$ minishift start
-- Starting profile 'minishift'
-- Checking if https://github.com is reachable (using proxy: "No") ... OK
```

### PR DESCRIPTION
**Pull request by LalatenduMohanty**
Fixes #2284  and #2292 

Signed-off-by: Lalatendu Mohanty <lmohanty@redhat.com>

### CODE REVIEW

**Comment by LalatenduMohanty** [comment]
retest please.

**Comment by LalatenduMohanty** [comment]
retest please

**Comment by LalatenduMohanty** [comment]
retest this please

File: cmd/minishift/cmd/start_preflight.go [hunk 1]
```diff
@@ -49,24 +49,18 @@ const (
...
+	// This is for security reasons as proxy URLs might contain credentials they should not
 	// always be shown.
-	if !glog.V(5) {
-		if os.Getenv("HTTPS_PROXY") == "" {
-			proxy = "No"
-		} else {
-			proxy = "Yes"
-		}
+	if proxyConfig.HttpsProxy() != "" || proxyConfig.HttpProxy() != "" {
+		fmt.Println("-- Using proxy for the setup")
```
Comment (LalatenduMohanty) [inline cmd/minishift/cmd/start_preflight.go]: I am having second thoughts about why we want to keep this preflight check as it is not doing any check.

**Comment by svenstaro** [comment]
I think this should still give you the option to print the proxy in case you give -v5

File: cmd/minishift/cmd/start_preflight.go (hunk 1 above)
Comment (budhrg) [reply cmd/minishift/cmd/start_preflight.go]: Agree :+1: 
This message is important when we use proxy flag in start.

Comment (anjannath) [reply cmd/minishift/cmd/start_preflight.go]: I think i got it, you want to put it somewhere else, as this is not a check and just letting the user know if proxy is used or not..

Comment (LalatenduMohanty) [reply cmd/minishift/cmd/start_preflight.go]: @anjannath Right, there is no reason it should be in preflight check.

**Comment by LalatenduMohanty** [comment]
@svenstaro @budhrg @anjannath Changed the code and moved the code in to handleProxies() in start.go. Please take a look.

**Comment by svenstaro** [comment]
LGTM :+1:

File: cmd/minishift/cmd/start.go [hunk 2]
```diff
@@ -292,10 +293,16 @@ func handleProxies() *util.ProxyConfig {
 		// proxy settings are properly passed to cluster up we need to explicitly set the values.
 		if proxyConfig.HttpProxy() != "" {
 			viper.Set(cmdUtil.HttpProxy, proxyConfig.HttpProxy())
+			if glog.V(5) {
+				fmt.Println(fmt.Sprintf("\tUsing http proxy: %s", proxyConfig.HttpProxy()))
+			}
```
Comment (praveenkumar) [inline cmd/minishift/cmd/start.go]: 👍

File: cmd/minishift/cmd/start.go [hunk 3]
```diff
@@ -292,10 +293,16 @@ func handleProxies() *util.ProxyConfig {
...
 		}
 
 		if proxyConfig.HttpsProxy() != "" {
 			viper.Set(cmdUtil.HttpsProxy, proxyConfig.HttpsProxy())
+			if glog.V(5) {
+				fmt.Println(fmt.Sprintf("\tUsing https proxy: %s", proxyConfig.HttpsProxy()))
+			}
```
Comment (praveenkumar) [inline cmd/minishift/cmd/start.go]: 👍

**Pull request merged**